src/service/routing/spec.clj

```diff
@@ -0,0 +1,67 @@
+(ns service.routing.spec
+  (:require [clojure.spec.gen.alpha :as gen]
+            [clojure.spec.alpha :as s]
+            [compojure.api.sweet :refer [context GET resource]]
+            [ring.util.http-response :refer [ok]]
+            [spec-tools.spec :as spec]
+            [service.routing.graph.generators :as g]
+            [service.routing.directions :as dir]))
+
+; (s/def :arc/kind (set (keys osm/speeds)))
+; (s/def :arc/length (s/and int? pos?))
+; (s/def :arc/dst int?)
+; (s/def :arc/src int?)
+
+(s/def ::lon (s/and number? #(<= -180 % 180)))
+(s/def ::lat (s/and number? #(<= -90 % 90)))
+; (s/def :node/arcs (s/coll-of :graph/arc :kind list?))
+; (s/def :node/out-arcs :node/arcs)
+; (s/def :node/in-arcs :node/arcs)
+
+; (s/def :graph/arc (s/and (s/keys :req-un [:arc/src :arc/dst :arc/length :arc/kind])
+;                          (fn [arc] (not= (:src arc) (:dst arc)))))
+; (s/def :graph/node (s/and (s/keys :req-un [:node/lat :node/lon :node/out-arcs :node/in-arcs])
+;                           (fn [node] (or (not-empty (:out-arcs node)) (not-empty (:in-arcs node))))))
+; (s/def :int-map/graph (s/map-of int? :graph/node))
+
+(s/def ::x spec/int?)
+(s/def ::y spec/int?)
+(s/def ::total spec/int?)
+(s/def ::total-map (s/keys :req-un [::total]))
+
+; ;(s/def ::vnum3 (s/coll-of number? :kind vector? :count 3 :distinct true :into #{}))
+(s/def ::direction (s/keys :req [::code ::waypoints ::routes]
```

Comment:
I think it should be `req-un` here, otherwise the whole key (namespace included) would have to be in the map